app/src/main/java/org/mozilla/fenix/components/Analytics.kt

```diff
@@ -117,12 +118,15 @@ class Analytics(
                 // from the app unless the user does so from a UI control.
                 // However, the user may have opt-ed out of mako experiments already, so
                 // we should respect that setting here.
-                val enabled = context.settings().isExperimentationEnabled
+                val enabled = context.components.strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
+                    context.settings().isExperimentationEnabled
+                }
                 if (!enabled) {
                     globalUserParticipation = enabled
                 }
-                context.settings().isExperimentationEnabled = globalUserParticipation
-
+            }
+        }.apply {
+            if (FeatureFlags.nimbusExperiments) {
```

Comment:
I broke this up, just to keep things clean.

Reply:
👍 Since we now know that the Nimbus constructor is quite expensive (even though it doesn't trigger any IO), I'd like to suggest we make this `nullable` and only create `Nimbus` if we need it to address one of the performance issues. Alternative we could return a "NoopNimbus" instance (cheap to create) which would make the consuming code cleaner.

```kotlin
val experiments: Nimbus? by lazyMonitored {
    val url: String? = BuildConfig.NIMBUS_ENDPOINT
    val serverSettings = if (!url.isNullOrBlank()) {
        NimbusServerSettings(url = Uri.parse(url))
    } else {
        null
    }

    if (FeatureFlags.nimbusExperiments) {
        Nimbus(context, serverSettings).apply {
            val enabled = context.components.strictMode.resetAfter(StrictMode.allowThreadDiskReads()) {
                context.settings().isExperimentationEnabled
            }
            if (!enabled) {
                globalUserParticipation = enabled
            }
            // Nimbus should look after downloading experiment definitions from remote settings
            // on another thread, and making sure we don't hit the server each time we start.
            updateExperiments()
        }
    } else {
        null // or NoopNimbus() :)
    }
}
```

Reply:
This would buy us some time and we can also uplift to beta so we don't take a performance hit there....don't need to decide now.